web/packages/shared/utils/error.ts

```diff
@@ -30,3 +30,7 @@ export function getErrMessage(err: unknown) {
 
   return message;
 }
+
+export function assertUnreachable(x: never): never {
+  throw new Error(`Unhandled case: ${x}`);
+}
```

Comment:
I think we can create a new file for this. It is not a function that checks error types.